src/Workspaces/CSharp/Portable/Extensions/SemanticModelExtensions.cs

```diff
@@ -203,6 +203,38 @@ private static bool CanBindToken(SyntaxToken token)
         public static string GenerateNameForExpression(
             this SemanticModel semanticModel, ExpressionSyntax expression, bool capitalize = false)
         {
+
+            // Check if expression is ArgumentSyntax. If we have an expression that
+            // is an argument, return the name of the parameter.
+            var parent = expression.Parent as ArgumentSyntax;
+            if (parent != null)
+            {
+                if (parent.NameColon != null)
+                {
+                    return parent.NameColon.Name.Identifier.ValueText.ToCamelCase();
+                }
+                else
+                {
+                    var argumentList = parent.Parent as ArgumentListSyntax;
+                    if (argumentList != null)
+                    {
+                        var argumentIndex = argumentList.Arguments.IndexOf(parent);
+                        var argumentInfo = GetArgumentInfo(semanticModel, argumentList.Arguments[argumentIndex]);
+                        if (argumentInfo.Parameter != null)
+                        {
+                            if (!argumentInfo.Parameter.IsParams)
+                            {
+                                return argumentInfo.Parameter.Name.ToCamelCase();
+                            }
+                            else if (!expression.IsAnyLiteralExpression())
+                            {
+                                return argumentInfo.Parameter.Name.ToCamelCase();
```

Comment:
It might be worth dogfooding the refactoring you're fixing on `argumentInfo.Parameter` :smile: